network/nodeTracker.py

```diff
@@ -50,8 +50,14 @@ def _add_data_point(self, data_point, output_file):
 	def _count_roles(nodes):
 		"""
 		Counts the number of nodes for each role.
+		Handles both Symbol and Nem nodes.
 		"""
 
+		# Process Nem nodes or empty nodes
+		if not nodes or 'roles' not in nodes[0]:
+			return {'total': len(nodes)}
```

Comment:
This is assuming ``nodes`` is not None?  I thinks ``len(nodes)`` will fail if nodes is None.